zaza/model.py

```diff
@@ -1740,3 +1741,81 @@ def attach_resource(application, resource_name, resource_path):
     cmd = ["juju", "attach-resource", "-m", juju_model,
            application, "{}={}".format(resource_name, resource_path)]
     subprocess.check_call(cmd)
+
+
+async def run_on_machine(machine, command, model_name=None, timeout=None):
```

Comment:
could you rename this async_run_on_machine  keeping it inline with the other async funcs in this module